plugins/modules/packaging/os/pulp_repo.py

```diff
@@ -688,6 +794,9 @@ def main():
             # we set the whole config at the same time.
             if not server.compare_repo_importer_config(
                 repo,
+                components=components,
```

Comment:
Did you check whether idempotency / updating actually works? Both for `rpm` and `deb` repositories?